api/src/templates/setup/setup.html

```diff
@@ -0,0 +1,163 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <title>{{ title }}</title>
+  <script>
+    const locale = "{{ locale }}";
+    const getListElem = () => document.getElementById('actions');
+    const clearList = () => {
+      const actionsElem = getListElem();
+      while (actionsElem.firstChild) {
+        actionsElem.removeChild(actionsElem.firstChild);
+      }
+    };
+
+    const createProgressListItem = (action) => {
+      const liElem = document.createElement('li');
+      liElem.className = 'step';
+
+      const loader = document.createElement('div');
+      let loaderClass;
+      if (action.completed) {
+        loaderClass = 'done';
+      } else if (action.started) {
+        loaderClass = 'loading';
+      }
+
+      loader.className = `loader ${loaderClass}`;
+      liElem.appendChild(loader);
+
+      const span = document.createElement('span');
+      span.textContent = action.text;
+      liElem.appendChild(span);
+
+      return liElem;
+    };
+
+    const populateList = (progress) => {
+      const actionsElem = getListElem();
+
+      progress.forEach((action) => {
+        if (!action.display) {
+          return;
+        }
+
+        const listElement = createProgressListItem(action);
+        actionsElem.appendChild(listElement);
+      });
+    }
+
+    const getProgress = async () => {
+
+      const result = await fetch('/progress', { headers: { 'accept-language': locale } });
```

Comment:
I think it would be cleaner to render all the HTML with all stages and translations using lodash templating, and then use this function to just toggle the CSS classes. That'll mean less login in JS and more handed off the the templating engine.

Reply:
I would have liked to have the steps in a single place instead of two. But I'll give this a shot :)